### PR TITLE
feat!: [BREAKING CHANGE]: Use `extra` instead of root

### LINKED ISSUE #581: [🐛] EAS build failing because of missing ios_app_id in app.json
### What happened?

I'm building using `eas build -p ios`, and I'm getting this error:

```
❌  error: ios_app_id key not found in react-native-google-mobile-ads key in app.json. App will crash without it.
```

Consider that this error is suddenly happening after I upgraded to Expo SDK 51.

```
⚠️  ld: ignoring duplicate libraries: '-lc++'
› Generating debug NoFilter » NoFilter.app.dSYM
› Executing NoFilter » Bundle React Native code and images
    Bundler cache is empty, rebuilding (this may take a minute)
› Executing NoFilter » [CP] Copy Pods Resources
› Executing NoFilter » Upload Debug Symbols to Sentry
› Executing NoFilter » [CP] Embed Pods Frameworks
› Executing NoFilter » [CP-User] [RNFB] Core Configuration
      A firebase.json file was not found, whilst this file is optional it is recommended to include it to configure firebase services in React Native Firebase.
› Executing NoFilter » [CP-User] [RNGoogleMobileAds] Configuration

❌  error: ios_app_id key not found in react-native-google-mobile-ads key in app.json. App will crash without it.

⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » NoFilter/NoFilter » Build Phases » 'Upload Debug Symbols to Sentry'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » NoFilter/NoFilter » Build Phases » '[CP-User] [RNFB] Core Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
⚠️  Script has ambiguous dependencies causing it to run on every build.
   To fix, go to: Xcode » NoFilter/NoFilter » Build Phases » '[CP-User] [RNGoogleMobileAds] Configuration'
   Either: Uncheck "Based on dependency analysis", or select output files to trigger the script
    Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'React-RuntimeApple' fr
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	PhaseScriptExecution [CP-User]\ [RNGoogleMobileAds]\ Configuration /Users/expo/Library/Developer/Xcode/DerivedData/NoFilter-fjavmkobuxfkzfckwbwnupmjterw/Build/Intermediates.noindex/ArchiveIntermediates/NoFilter/IntermediateBuildFilesPath/NoFilter.build/Release-iphoneos/NoFilter.build/Script-F562EEB2DBB1EB63B1F226D6.sh (in target 'NoFilter' from project 'NoFilter')
▸ (1 failure)
** ARCHIVE FAILED **
The following build commands failed:
	PhaseScriptExecution [CP-User]\ [RNGoogleMobileAds]\ Configuration /Users/expo/Library/Developer/Xcode/DerivedData/NoFilter-fjavmkobuxfkzfckwbwnupmjterw/Build/Intermediates.noindex/ArchiveIntermediates/NoFilter/IntermediateBuildFilesPath/NoFilter.build/Release-iphoneos/NoFilter.build/Script-F562EEB2DBB1EB63B1F226D6.sh (in target 'NoFilter' from project 'NoFilter')
(1 failure)

Exit status: 65
```

Now, the weird thing is that I have, and I always had (it was working before) the `ios_app_id` in my `app.json`. Like this:

```json
{
  "expo": {
    ...
  },
  "react-native-google-mobile-ads": {
    "android_app_id": "ca-app-pub-3342214521741372~6205894661",
    "ios_app_id": "ca-app-pub-3342214521741372~3029422209",
    "user_tracking_usage_description": "This identifier will be used to deliver personalized ads to you",
    "sk_ad_network_items": [
      "cstr6suwn9.skadnetwork",
      "4fzdc2evr5.skadnetwork",
      "4pfyvq9l8r.skadnetwork",
      "2fnua5tdw4.skadnetwork",
      "ydx93a7ass.skadnetwork",
      "5a6flpkh64.skadnetwork",
      "p78axxw29g.skadnetwork",
      "v72qych5uu.skadnetwork",
      "ludvb6z3bs.skadnetwork",
      "cp8zw746q7.skadnetwork",
      "3sh42y64q3.skadnetwork",
      "c6k4g5qg8m.skadnetwork",
      "s39g8k73mm.skadnetwork",
      "3qy4746246.skadnetwork",
      "f38h382jlk.skadnetwork",
      "hs6bdukanm.skadnetwork",
      "v4nxqhlyqp.skadnetwork",
      "wzmmz9fp6w.skadnetwork",
      "yclnxrl5pm.skadnetwork",
      "t38b2kh725.skadnetwork",
      "7ug5zh24hu.skadnetwork",
      "gta9lk7p23.skadnetwork",
      "vutu7akeur.skadnetwork",
      "y5ghdn5j9k.skadnetwork",
      "n6fk4nfna4.skadnetwork",
      "v9wttpbfk9.skadnetwork",
      "n38lu8286q.skadnetwork",
      "47vhws6wlr.skadnetwork",
      "kbd757ywx3.skadnetwork",
      "9t245vhmpl.skadnetwork",
      "eh6m2bh4zr.skadnetwork",
      "a2p9lx4jpn.skadnetwork",
      "22mmun2rn5.skadnetwork",
      "4468km3ulz.skadnetwork",
      "2u9pt9hc89.skadnetwork",
      "8s468mfl3y.skadnetwork",
      "klf5c3l5u5.skadnetwork",
      "ppxm28t8ap.skadnetwork",
      "ecpz2srf59.skadnetwork",
      "uw77j35x4d.skadnetwork",
      "pwa73g5rt2.skadnetwork",
      "mlmmfzh3r3.skadnetwork",
      "578prtvx9j.skadnetwork",
      "4dzt52r2t5.skadnetwork",
      "e5fvkxwrpn.skadnetwork",
      "8c4e2ghe7u.skadnetwork",
      "zq492l623r.skadnetwork",
      "3rd42ekr43.skadnetwork",
      "3qcr597p9d.skadnetwork"
    ]
  }
}
```

### Platforms

iOS, but have not tested behavior on Android

### React Native Info

```Shell
Consider that this build is being made on Expo EAS servers. Not in my local env.


System:
  OS: macOS 12.7.5
  CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
  Memory: 118.20 MB / 8.00 GB
  Shell:
    version: 5.8.1
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.1
    path: ~/.nvm/versions/node/v20.11.1/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.yarn/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.1/bin/npm
  Watchman:
    version: 2024.04.15.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.2 AI-232.10300.40.2321.11668458
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.1
    wanted: 0.74.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Are your using Typescript?

- [ ] My project is using Typescript

### package.json

```JSON
{
  "name": "no-filter-app",
  "version": "19.72.0",
  "private": true,
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@google-cloud/translate": "^6.1.0",
    "babel-eslint": "^10.1.0",
    "eslint": "^8.22.0",
    "eslint-plugin-react": "^7.31.0",
    "eslint-plugin-react-native": "^4.0.0",
    "patch-package": "^6.4.7",
    "postinstall-postinstall": "^2.1.0"
  },
  "main": "./node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start --dev-client",
    "generate-i18n": "node ./generate-i18n.js",
    "update-changelogs": "node ./update-changelogs.js",
    "postinstall": "patch-package",
    "build:develop": "eas build --profile development -p ios"
  },
  "dependencies": {
    "@amplitude/analytics-react-native": "^1.4.8",
    "@expo/react-native-action-sheet": "^4.1.0",
    "@react-native-async-storage/async-storage": "1.23.1",
    "@react-native-community/netinfo": "11.3.1",
    "@react-native-firebase/analytics": "^19.2.2",
    "@react-native-firebase/app": "^19.2.2",
    "@react-native-google-signin/google-signin": "^11.0.1",
    "@sentry/react-native": "~5.22.0",
    "country-emoji": "^1.5.6",
    "duix": "^3.0.3",
    "expo": "~51.0.8",
    "expo-apple-authentication": "~6.4.1",
    "expo-application": "~5.9.1",
    "expo-background-fetch": "~12.0.1",
    "expo-build-properties": "~0.12.1",
    "expo-clipboard": "~6.0.3",
    "expo-constants": "~16.0.1",
    "expo-dev-client": "~4.0.14",
    "expo-device": "~6.0.2",
    "expo-file-system": "~17.0.1",
    "expo-font": "~12.0.5",
    "expo-image": "~1.12.9",
    "expo-image-picker": "~15.0.5",
    "expo-linking": "~6.3.1",
    "expo-localization": "~15.0.3",
    "expo-location": "~17.0.1",
    "expo-media-library": "~16.0.3",
    "expo-notifications": "~0.28.3",
    "expo-quick-actions": "^2.0.0",
    "expo-splash-screen": "~0.27.4",
    "expo-store-review": "~7.0.2",
    "expo-task-manager": "~11.8.1",
    "expo-tracking-transparency": "~4.0.2",
    "expo-updates": "~0.25.14",
    "i18n-js": "^3.7.1",
    "lodash": "^4.17.21",
    "react": "18.2.0",
    "react-native": "0.74.1",
    "react-native-fbsdk-next": "^12.2.0",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-google-mobile-ads": "^13.6.0",
    "react-native-maps": "1.14.0",
    "react-native-purchases": "^7.27.2",
    "react-native-reanimated": "~3.10.1",
    "react-native-reanimated-carousel": "4.0.0-alpha.12",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "4.10.1",
    "react-native-svg": "15.2.0",
    "react-native-uuid": "^2.0.2",
    "rn-prompt": "^1.0.4",
    "sanitize-html": "^2.13.0",
    "sunrise-sunset-js": "^2.2.1",
    "use-interval": "^1.4.0"
  }
}
```

### app.json

```JSON
{
  "expo": {
    "name": "NoFilter",
    "version": "19.72.0",
    "description": "NoFilter helps photographers and travelers to find the best photo spots nearby, and all around the world",
    "icon": "src/styles/imgs/nofilter-icon.png",
    "orientation": "portrait",
    "primaryColor": "#ffffff",
    "splash": {
      "backgroundColor": "#000",
      "resizeMode": "cover",
      "image": "src/styles/imgs/splash-screen-ph.png"
    },
    "notification": {
      "icon": "src/styles/imgs/noti-icon.png"
    },
    "locales": {
      "ar": "./locales/ar.json",
      "bg": "./locales/bg.json",
      "ca": "./locales/ca.json",
      "cs": "./locales/cs.json",
      "de": "./locales/de.json",
      "en": "./locales/en.json",
      "es": "./locales/es.json",
      "fr": "./locales/fr.json",
      "hi": "./locales/hi.json",
      "hr": "./locales/hr.json",
      "id": "./locales/id.json",
      "is": "./locales/is.json",
      "it": "./locales/it.json",
      "ja": "./locales/ja.json",
      "ms": "./locales/ms.json",
      "nl": "./locales/nl.json",
      "pl": "./locales/pl.json",
      "pt": "./locales/pt.json",
      "ru": "./locales/ru.json",
      "tr": "./locales/tr.json",
      "vi": "./locales/vi.json",
      "zh": "./locales/zh.json"
    },
    "slug": "xx",
    "scheme": "nofilter",
    "jsEngine": "hermes",
    "plugins": [
      [
        "@sentry/react-native/expo",
        {
          "organization": "xx",
          "project": "xxx",
          "setCommits": true
        }
      ],
      "@react-native-google-signin/google-signin",
      [
        "expo-media-library",
        {
          "photosPermission": "We use your library when you want to upload a photo to your feed, or a new profile picture",
          "savePhotosPermission": "We use your library when you want to upload a photo to your feed, or a new profile picture",
          "isAccessMediaLocationEnabled": true
        }
      ],
      [
        "react-native-fbsdk-next",
        {
          "appID": "xxx",
          "clientToken": "xxx",
          "displayName": "NoFilter",
          "advertiserIDCollectionEnabled": false,
          "autoLogAppEventsEnabled": false,
          "isAutoInitEnabled": false,
          "iosUserTrackingPermission": false
        }
      ],
      "./plugins/",
      "@react-native-firebase/app",
      [
        "expo-build-properties",
        {
          "ios": {
            "useFrameworks": "static"
          }
        }
      ],
      "expo-localization",
      [
        "expo-tracking-transparency",
        {
          "userTrackingPermission": "This identifier will be used to deliver personalized ads to you"
        }
      ],
      [
        "expo-font",
        {
          "fonts": [
            "src/styles/fonts/Montserrat/Montserrat-Regular.ttf",
            "src/styles/fonts/Montserrat/Montserrat-Medium.ttf",
            "src/styles/fonts/Montserrat/Montserrat-SemiBold.ttf"
          ]
        }
      ],
      [
        "expo-quick-actions",
        {
          "androidIcons": {
            "help_icon": {
              "foregroundImage": "src/styles/imgs/android-adaptive-foreground.png",
              "backgroundColor": "#000000"
            }
          },
          "iosActions": [
            {
              "title": "Wait! Don't delete me!",
              "subtitle": "Tell me what happened",
              "icon": "symbol:person.crop.circle.badge.questionmark",
              "id": "0",
              "params": {
                "href": "/about-to-remove"
              }
            }
          ]
        }
      ]
    ],
    "updates": {
      "enabled": false,
      "url": "xxxx"
    },
    "ios": {
      "buildNumber": "19.72.0",
      "supportsTablet": true,
      "googleServicesFile": "./GoogleService-Info.plist",
      "bundleIdentifier": "app.no-filter.nofilter",
      "appStoreUrl": "https://itunes.apple.com/us/app/nofilter-photo-spots/id1445583976",
      "usesAppleSignIn": true,
      "config": {
        "usesNonExemptEncryption": false,
        "googleMapsApiKey": "xxxx"
      },
      "infoPlist": {
        "NSUserTrackingUsageDescription": "This identifier will be used to deliver personalized ads to you",
        "NSLocationWhenInUseUsageDescription": "We use your location to see how far you are from spots, and also to show you your position in the map",
        "NSPhotoLibraryUsageDescription": "We use your library when you want to upload a photo to your feed, or a new profile picture",
        "NSPhotoLibraryAddUsageDescription": "We use your library when you want to upload a photo to your feed, or a new profile picture",
        "CFBundleAllowMixedLocalizations": true,
        "SKAdNetworkItems": [
          {
            "SKAdNetworkIdentifier": "xxxx"
          },
          {
            "SKAdNetworkIdentifier": "xxx"
          }
        ],
        "LSApplicationQueriesSchemes": [
          "instagram",
          "twitter"
        ]
      }
    },
    "android": {
      "versionCode": 483,
      "package": "app.no_filter.nofilter",
      "playStoreUrl": "https://play.google.com/store/apps/details?id=app.no_filter.nofilter",
      "icon": "src/styles/imgs/nofilter-icon-android.png",
      "adaptiveIcon": {
        "foregroundImage": "src/styles/imgs/android-adaptive-foreground.png",
        "backgroundColor": "#000000"
      },
      "config": {
        "googleMaps": {
          "apiKey": "xxxx"
        }
      },
      "googleServicesFile": "./google-services.json",
      "permissions": [
        "ACCESS_COARSE_LOCATION",
        "ACCESS_FINE_LOCATION",
        "READ_EXTERNAL_STORAGE",
        "WRITE_EXTERNAL_STORAGE",
        "ACCESS_MEDIA_LOCATION",
        "android.permission.READ_EXTERNAL_STORAGE",
        "android.permission.WRITE_EXTERNAL_STORAGE",
        "android.permission.ACCESS_MEDIA_LOCATION",
        "android.permission.INTERNET",
        "com.google.android.gms.permission.AD_ID"
      ],
      "blockedPermissions": [
        "android.permission.CAMERA",
        "android.permission.RECORD_AUDIO"
      ]
    },
    "platforms": [
      "android",
      "ios"
    ],
    "extra": {
      "eas": {
        "projectId": "xxxx"
      }
    },
    "runtimeVersion": {
      "policy": "appVersion"
    },
    "owner": "brodanoel"
  },
  "react-native-google-mobile-ads": {
    "android_app_id": "ca-app-pub-3342214521741372~6205894661",
    "ios_app_id": "ca-app-pub-3342214521741372~3029422209",
    "user_tracking_usage_description": "This identifier will be used to deliver personalized ads to you",
    "sk_ad_network_items": [
      "cstr6suwn9.skadnetwork",
      "4fzdc2evr5.skadnetwork",
      "4pfyvq9l8r.skadnetwork",
      "2fnua5tdw4.skadnetwork",
      "ydx93a7ass.skadnetwork",
      "5a6flpkh64.skadnetwork",
      "p78axxw29g.skadnetwork",
      "v72qych5uu.skadnetwork",
      "ludvb6z3bs.skadnetwork",
      "cp8zw746q7.skadnetwork",
      "3sh42y64q3.skadnetwork",
      "c6k4g5qg8m.skadnetwork",
      "s39g8k73mm.skadnetwork",
      "3qy4746246.skadnetwork",
      "f38h382jlk.skadnetwork",
      "hs6bdukanm.skadnetwork",
      "v4nxqhlyqp.skadnetwork",
      "wzmmz9fp6w.skadnetwork",
      "yclnxrl5pm.skadnetwork",
      "t38b2kh725.skadnetwork",
      "7ug5zh24hu.skadnetwork",
      "gta9lk7p23.skadnetwork",
      "vutu7akeur.skadnetwork",
      "y5ghdn5j9k.skadnetwork",
      "n6fk4nfna4.skadnetwork",
      "v9wttpbfk9.skadnetwork",
      "n38lu8286q.skadnetwork",
      "47vhws6wlr.skadnetwork",
      "kbd757ywx3.skadnetwork",
      "9t245vhmpl.skadnetwork",
      "eh6m2bh4zr.skadnetwork",
      "a2p9lx4jpn.skadnetwork",
      "22mmun2rn5.skadnetwork",
      "4468km3ulz.skadnetwork",
      "2u9pt9hc89.skadnetwork",
      "8s468mfl3y.skadnetwork",
      "klf5c3l5u5.skadnetwork",
      "ppxm28t8ap.skadnetwork",
      "ecpz2srf59.skadnetwork",
      "uw77j35x4d.skadnetwork",
      "pwa73g5rt2.skadnetwork",
      "mlmmfzh3r3.skadnetwork",
      "578prtvx9j.skadnetwork",
      "4dzt52r2t5.skadnetwork",
      "e5fvkxwrpn.skadnetwork",
      "8c4e2ghe7u.skadnetwork",
      "zq492l623r.skadnetwork",
      "3rd42ekr43.skadnetwork",
      "3qcr597p9d.skadnetwork"
    ]
  }
}
```

### ios/Podfile

```Ruby
Not visible on Expo
```

### android/build.gradle

```Groovy
Not visible on Expo
```

### android/app/build.gradle

```Groovy
Not visible on Expo
```

### android/settings.gradle

```Groovy
Not visible on Expo
```

### AndroidManifest.xml

```XML
Not visible on Expo
```

### PR DESCRIPTION
**Pull request by BrodaNoel**
### Description
This PR tries to fix https://github.com/invertase/react-native-google-mobile-ads/issues/581. All the description of this problem is defined there, but, basically, since Expo SDK 51, it won't work if we try to get the `react-native-google-mobile-ads` key from the root of `app.json`. We need to get it from `root.extra` (thus, we need to change this documentation as well)

Please don't kill me... I have no idea what I am doing in that file (it was actually 100% gpt-4o).

The behavior implemented is this:
First, tries to find the `react-native-google-mobile-ads` key inside `root > expo > extra`, then, if it doesn't find it, it tries in `root > extra` (because Expo Cli may merge it soon).

### Pending stuff

We'll need to also change the documentation of this library, asking the users to put `react-native-google-mobile-ads` inside `extra`, instead of directly inside the root.

### Related issues

Fixes #581 

### Release Summary

Breaking: Move the `react-native-google-mobile-ads` key in `app.json` from `root` to `root.extra`

### Checklist

- I read the [Contributor Guide](https://github.com/invertase/react-native-google-mobile-ads/blob/main/CONTRIBUTING.md)
  and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `__tests__e2e__`
  - [ ] `jest` tests added or updated in `__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [x] Yes
  - [ ] No

### Test Plan

I'm not sure if this works. I'm proposing this PR as a first initial version

---

Think `react-native-google-mobile-ads` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by BrodaNoel** [comment]
This will break Android builds.
Just use it as an idea of what we have to implement. Close it as soon as you get it

File: ios_config.sh [hunk 1]
```diff
@@ -62,7 +61,7 @@ function setPlistValue {
 
 function getJsonKeyValue () {
   if [[ ${_RN_ROOT_EXISTS} ]]; then
-    ruby -KU -e "require 'rubygems';require 'json'; output=JSON.parse('$1'); puts output[$_JSON_ROOT]['$2']"
+    ruby -KU -e "require 'rubygems';require 'json'; output=JSON.parse('$1'); puts output['expo']['extra']['react-native-google-mobile-ads']['$2']"
```
Comment (BrodaNoel) [inline ios_config.sh:64]: The problem here is that some people are already removing the `expo` property in the `app.json` file, so, this code should actually first check if it `root > extra > react-native-google-mobile-ads` is present, if not, check `root > expo > react-native-google-mobile-ads`

**Comment by BrodaNoel** [comment]
I'm not sure why, but Android is not working.
I'm trying to find some clues about it

File: android/app-json.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ for (int i = 0; i <= 3; i++) {
     else {
       configFile = new File(parentDir, fileNames[1])
       if (configFile.exists()) {
-        fileName = fileNames[0]
+        fileName = fileNames[1]
```
Comment (BrodaNoel) [inline android/app-json.gradle:24]: This is probably an old bug

File: android/app-json.gradle [hunk 3]
```diff
@@ -33,17 +33,11 @@ if (configFile != null && configFile.exists()) {
   Object json = null
 
   try {
-    // On windows, we need to escape path separators in the path before exec'ing shell
```
Comment (BrodaNoel) [inline android/app-json.gradle:36]: All comments removed by ChatGPT

**Comment by DoctorJohn** [comment]
> basically, since Expo SDK 51, it won't work

I'm not sure that's true. Even in the referenced issue some people say it doesn't work with SDK 50 either. From my understanding nothing relevant changed in expo but PR #517 broke loading of the `react-native-google-mobile-ads` config from `app.json` when there is also an `app.config.js` file present. PR #521 proposes a fix for this but also adds additional features which make it unlikely it will ever be merged.

I think we need to reevaluate how we want to insert this packages config into native code. Ideally we would have a config plugin for expo but that would conflict with the `ios_config.sh` and the `app-json.gradle` files which implement a similar mechanism.

**Comment by BrodaNoel** [comment]
@DoctorJohn actually there is something who was changed on SDK 51 who broke it. Read this: https://github.com/expo/fyi/blob/main/root-expo-object.md#migrating-the-config

**Comment by longtimeno-c** [comment]
Has the config changed at all in expo? or is app.json still setup same as written in documentation?

> 🎉 This issue has been resolved in version 14.0.0 🎉
> 
> The release is available on:
> 
> * [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-native-google-mobile-ads/v/14.0.0)
> * [GitHub release](https://github.com/invertase/react-native-google-mobile-ads/releases/tag/v14.0.0)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

**Comment by dylancom** [comment]
Scherm­afbeelding 2024-07-12 om 14 52 20

**Pull request closed without merging**